haddock-api/src/Haddock/Interface/Create.hs

```diff
@@ -165,6 +193,30 @@ createInterface mod_iface flags modMap instIfaceMap = do
   , ifaceWarningMap        = warningMap
   , ifaceTokenizedSrc      = Nothing -- TODO: Get this from the extended .hie-files.
   }
+  where
+    -- Note [Exporting built-in items]
+    --
+    -- Some items do not show up in their modules exports simply because Haskell
+    -- lacks the concrete syntax to represent such an export. We'd still like
+    -- these to show up in docs, so we manually patch on some extra exports for a
+    -- small number of modules:
+    --
+    --   * "GHC.Prim" should export @(->)@
+    --   * "GHC.Types" should export @[]([], (:))@ and @(~)@
+    --   * "Prelude" should export @(->)@ and @[]([], (:))@
+    --   * "Data.Tuple" should export tuples up to arity 15
+    --
+    listAvail = [ AvailTC listTyConName
+                          [listTyConName, nilDataConName, consDataConName]
+                          [] ]
+    funAvail  = [ AvailTC funTyConName [funTyConName] [] ]
+    eqAvail   = [ AvailTC eqTyConName [eqTyConName] [] ]
+    tupsAvail = [ AvailTC tyName [tyName, datName] []
+                | i<-[0..15]
```

Comment:
Can you add a comment on why you chose arities 0 to 15 here? I've seen `mAX_TUPLE_SIZE` in ghc but I don't think it's a better limit than 15.

Personally I'm a bit surprised that we have tuples of arity 0 and 1 but I may have simply missed the news.